### PR TITLE
Update openScmp to read file structure

### PR DESCRIPTION
**Pull request by SakshiS-harma**
This is the second half of the changes to fix https://github.com/microsoft/azuredatastudio/issues/20332.

### CODE REVIEW

**Review by Benjin — APPROVED** [review]

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOpenScmpOperation.cs [hunk 1]
```diff
@@ -127,17 +129,38 @@ private SchemaCompareEndpointInfo GetEndpointInfo(bool source, SchemaCompareEndp
                         {
                             if (node.Parent.Name.ToString().Contains(searchingFor))
                             {
-                                endpointInfo.ConnectionDetails = SchemaCompareService.ConnectionServiceInstance.ParseConnectionString(node.Value);
-                                endpointInfo.ConnectionDetails.ConnectionString = node.Value;
-                                endpointInfo.DatabaseName = endpointInfo.ConnectionDetails.DatabaseName;
-                                endpointInfo.EndpointType = SchemaCompareEndpointType.Database;
+                                if(isProjectEndpoint)
```
Comment (kisantia) [inline src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOpenScmpOperation.cs:132]: it's a little hard to follow this with both the project and database endpoint logic mixed together. Can they be split up so it's easier to follow and if there's some common code, that could be split into a helper function?

File: test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/SchemaCompare/SchemaCompareServiceTests.cs [hunk 2]
```diff
@@ -1865,6 +1865,8 @@ private void ValidateResultEndpointInfo(SchemaCompareEndpoint originalEndpoint,
             {
                 SchemaCompareProjectEndpoint projectEndpoint = originalEndpoint as SchemaCompareProjectEndpoint;
                 Assert.AreEqual(projectEndpoint.ProjectFilePath, resultEndpoint.ProjectFilePath);
+                Assert.AreEqual(resultEndpoint.ExtractTarget, DacExtractTarget.ObjectType);
```
Comment (kisantia) [inline test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/SchemaCompare/SchemaCompareServiceTests.cs:1868]: also add strings to describe these asserts so it's easier to know what failed if the tests fail

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOpenScmpOperation.cs [hunk 3]
```diff
@@ -146,6 +169,25 @@ private SchemaCompareEndpointInfo GetEndpointInfo(bool source, SchemaCompareEndp
             return endpointInfo;
         }
 
+
+        /**
+         * Function to map folder structure string to enum
+         * @param inputTarget folder structure in string
+         * @returns folder structure in enum format
+         */
+        private DacExtractTarget? mapExtractTargetEnum(string inputTarget)
```
Comment (Benjin) [inline src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOpenScmpOperation.cs:178]: styling: C# does capitalized functions, even for private.

**Review by Benjin — APPROVED** [review]

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOpenScmpOperation.cs (hunk 1 above)
Comment (SakshiS-harma) [reply src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOpenScmpOperation.cs:127]: Split- done.

**Review by kisantia — APPROVED** [review]
